run_tests.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+
+# This file detects and run all tests
+# USAGE:
+#
+# * Run all tests:
+# $ ./run_tests.sh
+#
+# * Run some tests.sh:
+# $ ./run_tests.sh test_bar test_foo
+
+
+assertBuildFailure() {
+    set +e
+
+    bazel build "$1"
+    if [ $? -eq 0 ]; then
+        echo "bazel build {$1} should have failed."
+        exit 1
+    fi
+
+    set -e
+}
+
+# Color if we are in a terminal, no color otherwise
+if [ -t 1 ]
+then
+    NC='\033[0m'
+    GREEN='\033[0;32m'
+    RED='\033[0;31m'
+fi
+
+run_test() {
+  set +e
+  SECONDS=0
+  TEST_ARG=$@
+  echo "running test $TEST_ARG"
+  RES=$($TEST_ARG 2>&1)
+  RESPONSE_CODE=$?
+  DURATION=$SECONDS
```

Comment:
Isn't `DURATION` always 0?

Reply:
Ah, I didn't know about this shell function.

Reply:
```
       SECONDS
              Each  time this parameter is referenced, the number of seconds since shell invocation is returned.  If
              a value is assigned to SECONDS, the value returned upon subsequent references is the number of seconds
              since  the  assignment plus the value assigned.  If SECONDS is unset, it loses its special properties,
              even if it is subsequently reset.
```

(from `man bash`)